### ISSUE TITLE
Does it make sense to switch to monaco-editor instead of monaco-editor-core

### ISSUE THREAD

**Issue by kaisalmen** [TypeFox/monaco-languageclient#337]
I was wondering if we should switch to the dependency of `monaco-editor-core` to `monaco-editor`.

Why? Because you cannot import both `monaco-editor` and `monaco-editor-core` in the same project/app. This blocks using features available from the main editor (like js support) and adding a new language + LS connection via this project, because both define the same things in global space.

As far as I understand it you can just important the "core" feature of `monaco-editor` like this:
```javascript
import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
```
and then you achieve the same as now, but you don't block the combined usage outlined above. Is this reasoning correct? WDYT? Does this have any bad implications I am overlooking?

**Comment by CGNonofr** [comment]
I agree!

Btw, since we only import it for types, It won't change anything to the generated code, will it?

We can switch on monaco-editor for the example tho, we'll at least have the syntax highlighting.

**Comment by CGNonofr** [comment]
The javascript won't change, but the types will changes indeed. I don't see a single reason why someone would use monaco-editor-core instead of monaco-editor

Btw importing `monaco-editor/esm/vs/editor/editor.api` instead of `monaco-editor` makes no difference here.

What is a little weird however is that monaco-editor or monaco-editor-core is not in the dependencies of monaco-languageclient (not event in peer dependencies)

**Comment by kaisalmen** [comment]
Otherwise, the TypeScript compile step must go 💥 because the import does not exist?

**Comment by kaisalmen** [comment]
> so ther can import it any way they want, it can be either monaco-editor or monaco-editor-core.

💡
 
> I needed to enable skipLibCheck for this reason IIRC

Ok, also good to know.

**Comment by kaisalmen** [comment]
Resolved with #340
